Source/WebKit/UIProcess/Inspector/WebInspectorUIProxy.cpp

```diff
@@ -759,6 +759,11 @@ void WebInspectorUIProxy::setDeveloperPreferenceOverride(WebCore::InspectorClien
             inspectedPage->setMockCaptureDevicesEnabledOverride(overrideValue);
 #endif // ENABLE(MEDIA_STREAM)
         return;
+
+    case InspectorClient::DeveloperPreference::NeedsSiteSpecificQuirks:
+        if (RefPtr inspectedPage = m_inspectedPage.get())
+            inspectedPage->protectedPreferences()->setNeedsSiteSpecificQuirks(overrideValue && overrideValue.value());
```

Comment:
rather than modify the preference directly, can we have this be an override like all the others so that when Web Inspector is closed we're able to go back to whatever behavior it had before?

Reply:
@dcrousso would there be a possibility for people on iOS to be able to disable Site Specific Quirks through a menu in addition to Web Inspector, in order to allow people testing sites on mobile but do not have access to a macbook.

Reply:
@karlcow That mechanism is controlled outside of WebKit code, so would need to be changed elsewhere.

Reply:
> rather than modify the preference directly, can we have this be an override like all the others so that when Web Inspector is closed we're able to go back to whatever behavior it had before?

That's a great idea! I'll do so and get a new patch up.

Reply:
Updated patch with this change, @dcrousso .

Reply:
@karlcow i coulda sworn there was a toggle for it somewhere in iOS/Safari settings, but i can't seem to find it

regardless, the general principle of Web Inspector is that anything changed while debugging should be undone as soon as Web Inspector is disconnected, with the only exceptions being HTML, CSS, and JS (i.e. changes to page content can persist but engine/browser adjustments must be undone)